code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/request/handler/GenericWhitelistedProxy.java

```diff
@@ -214,6 +217,21 @@ protected void generate(final ApiRequest request) throws IOException {
             throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
         }
 
+        boolean matchesAllowedPath = false;
+        if(isNoAuthProxy()) {
+            if (uri.getPath() != null && allowedPaths != null) {
```

Comment:
@ibuildthecloud the request URL has the prefix /v1/proxy, while the redirect URL is the actual URL requested by the UI.

example:
requestURL is = http://localhost:8080/v1/proxy/http://localhost:8090/v1-auth/saml/login

redirect is= http://localhost:8090/v1-auth/saml/login

So I am checking the redirect.